crates/oxc_language_server/src/main.rs

```diff
@@ -328,10 +328,19 @@ impl Backend {
     }
 
     async fn is_ignored(&self, uri: &Url) -> bool {
+        let Some(Some(root_uri)) = self.root_uri.get() else {
+            return false;
+        };
         let Some(ref gitignore_globs) = *self.gitignore_glob.lock().await else {
             return false;
         };
+
+        let root_path = PathBuf::from(root_uri.path());
         let path = PathBuf::from(uri.path());
+        // The file is not under current workspace
+        if !path.starts_with(root_path) {
```

Comment:
If allocation is removed, does it make sense to move this check before the ignore check?